### PR TITLE
Replace TlsStream type by using SslStream directly

### PR DESCRIPTION
**Pull request by rzikm**
Follow up after #106362.

The `TlsStream` type does not bring much useful behavior on top of plain `SslStream`, removing the type prevents issues like https://github.com/dotnet/runtime/issues/99276 stemming from missing delegating overloads from resurfacing in the future.

Summary of changes:
- Replace TlsStream usages with SslStream
- Inline parameters for SslProtocols and Cert validation callback on AuthenticateAs* callsites
- Removed TlsStream.cs

There was some extra plumbing necessary in case of System.Net.Requests where the code needs to manipulate the underlying NetworkStream (mainly accessing the Socket directly), so the original `NetworkStream` is being passed along as well where appropriate.

### CODE REVIEW

File: src/libraries/System.Net.Requests/src/System/Net/FtpControlStream.cs [hunk 1]
```diff
@@ -211,7 +227,7 @@ protected override void ClearState()
 
             _dataSocket = null;
             _passiveEndPoint = null;
-            _tlsStream = null;
+            _sslStream = null;
```
Comment (MihaZupan) [inline src/libraries/System.Net.Requests/src/System/Net/FtpControlStream.cs:230]:
```suggestion
            _sslStream = null;
            _dataStream = null;
```

I don't think it functionally matters though

**Review by MihaZupan — APPROVED** [review]
Good riddance, thank you

File: src/libraries/System.Net.Requests/src/System/Net/FtpDataStream.cs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
```
Comment (wfurt) [inline src/libraries/System.Net.Requests/src/System/Net/FtpDataStream.cs:4]: nit: do we need this?

**Review by wfurt — APPROVED** [review]
LGTM. Hopefully we will have whole release to bake it.

File: src/libraries/System.Net.Requests/src/System/Net/FtpDataStream.cs (hunk 2 above)
Comment (rzikm) [inline src/libraries/System.Net.Requests/src/System/Net/FtpDataStream.cs:4]:
```suggestion
```

**Pull request merged**